micro-rdk/src/esp32/exec.rs

```diff
@@ -2,55 +2,47 @@
 use crate::common::webrtc::exec::WebRtcExecutor;
 use async_executor::{LocalExecutor, Task};
 use futures_lite::{
-    future::{self},
+    future::{self, block_on},
     Future,
 };
-use std::rc::Rc;
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, Default)]
 /// This executor is local and bounded to the CPU that created it usually you would create it after spwaning a thread on a specific core
-pub struct Esp32Executor<'a> {
-    /// A local executor
-    executor: Rc<LocalExecutor<'a>>,
+pub struct Esp32Executor {}
+
+std::thread_local! {
+    static EX: LocalExecutor<'static> = LocalExecutor::new();
 }
 
-impl<'a> Esp32Executor<'a> {
+impl Esp32Executor {
     /// Return a new executor bounded to the current core.
     pub fn new() -> Self {
-        Esp32Executor {
-            executor: Rc::new(LocalExecutor::new()),
-        }
-    }
-    /// Spawn a future onto the local executor
-    pub fn spawn<T: 'a>(&self, future: impl Future<Output = T> + 'a) -> Task<T> {
-        self.executor.spawn(future)
+        Self {}
```

Comment:
All `Esp32Executor`s are now equivalent. Should this be `get` or similar now? Or `instance`? Are we planning to move to a world where there is a single executor?

Reply:
getting it will take us in lifetime nightmare :P